<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGL Test</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="NBodyPanel" name="nbodyPanel" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <property name="labelAlignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="useOpenCLLabel">
         <property name="text">
          <string>Use OpenCL:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QCheckBox" name="useOpenCLCheckBox">
         <property name="text">
          <string/>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="useGPULabel">
         <property name="text">
          <string>Use GPU:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="useGPUCheckBox">
         <property name="text">
          <string/>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="useTilesLabel">
         <property name="text">
          <string>Use Tiles:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QCheckBox" name="useTilesCheckBox">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="tileSizeLabel">
         <property name="text">
          <string>Tile Size:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="tileSizeSpinBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1024</number>
         </property>
         <property name="value">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="bodyCountLabel">
         <property name="text">
          <string>Body Count:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="bodyCountSpinBox">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>999999</number>
         </property>
         <property name="singleStep">
          <number>50</number>
         </property>
         <property name="value">
          <number>2048</number>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QPushButton" name="startButton">
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QPushButton" name="stopButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="QLabel" name="textEditLabel">
         <property name="text">
          <string>Time Steps Per Second:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="timeStepOutput">
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Courier New</family>
          </font>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="averageLabel">
         <property name="text">
          <string>Average Time Steps Per Second:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="averageLineEdit"/>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>NBodyPanel</class>
   <extends>QWidget</extends>
   <header>nbodypanel.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>startSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>515</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>stopSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>768</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>useOpenCLCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>useGPUCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>676</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>677</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>useOpenCLCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>useTilesCheckBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>671</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>703</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>useOpenCLCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>tileSizeSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>678</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>714</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>startSimulation()</slot>
  <slot>stopSimulation()</slot>
 </slots>
</ui>
